linkerd/stack/src/result.rs

```diff
@@ -0,0 +1,57 @@
+use futures::{future, prelude::*, ready};
+use linkerd2_error::Error;
+use std::task::{Context, Poll};
+
+#[derive(Copy, Clone, Debug)]
+pub struct ResultService<S, E>(Inner<S, E>);
+
+#[derive(Copy, Clone, Debug)]
+enum Inner<S, E> {
+    Ok(S),
+    Err(Option<E>),
+}
```

Comment:
nit, take it or leave it: isn't this just
```suggestion
pub struct ResultService<S, E>(Result<S, E>);
```